packages/convex-helpers/react/sessions.ts

```diff
@@ -98,18 +98,16 @@ export const SessionProvider: React.FC<{
   const useStorageOrDefault = useStorage ?? useSessionStorage;
   const [sessionId, setSessionId] = useStorageOrDefault(
     storeKey,
-    ssrFriendly ? SSR_DEFAULT : idGen()
+    ssrFriendly ? undefined : idGen()
   );
 
   const [initial, setInitial] = useState(true);
-  if (ssrFriendly) {
-    // Generate a new session ID on first load.
-    // This is to get around SSR issues with localStorage.
-    useEffect(() => {
-      if (sessionId === SSR_DEFAULT) setSessionId(idGen());
-      if (initial) setInitial(false);
-    }, [setSessionId, sessionId, setInitial, initial]);
-  }
+  // Generate a new session ID on first load.
+  // This is to get around SSR issues with localStorage.
+  useEffect(() => {
+    if (!sessionId || sessionId === SSR_DEFAULT) setSessionId(idGen());
```

Comment:
You're keeping the SSR_DEFAULT for backwards compatibility?

Reply:
yeah or as a backstop